comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go

```diff
@@ -232,85 +243,82 @@ func (ia *inventoryagent) fetchCoreAgentMetadata() {
 }
 
 func (ia *inventoryagent) fetchSecurityAgentMetadata() {
-	securityCfg := ia.getCorrectConfig("security-agent", ia.conf, fetchSecurityConfig, ia.conf)
+	securityCfg := ia.getCorrectConfig("security-agent", "runtime_security_config.enabled", ia.conf, fetchSecurityConfig)
 
 	ia.data["feature_cspm_enabled"] = securityCfg.GetBool("compliance_config.enabled")
 	ia.data["feature_cspm_host_benchmarks_enabled"] = securityCfg.GetBool("compliance_config.enabled") && securityCfg.GetBool("compliance_config.host_benchmarks.enabled")
 }
 
 func (ia *inventoryagent) fetchTraceAgentMetadata() {
-	traceCfg := ia.getCorrectConfig("trace-agent", ia.conf, fetchTraceConfig, ia.conf)
+	traceCfg := ia.getCorrectConfig("trace-agent", "apm_config.enabled", ia.conf, fetchTraceConfig)
 
 	ia.data["config_apm_dd_url"] = scrub(traceCfg.GetString("apm_config.apm_dd_url"))
 	ia.data["feature_apm_enabled"] = traceCfg.GetBool("apm_config.enabled")
 }
 
 func (ia *inventoryagent) fetchProcessAgentMetadata() {
-	processCfg := ia.getCorrectConfig("process-agent", ia.conf, fetchProcessConfig, ia.conf)
+	processCfg := ia.getCorrectConfig("process-agent", "process_config.process_collection.enabled", ia.conf, fetchProcessConfig)
 
 	ia.data["feature_process_enabled"] = processCfg.GetBool("process_config.process_collection.enabled")
 	ia.data["feature_processes_container_enabled"] = processCfg.GetBool("process_config.container_collection.enabled")
 	ia.data["feature_process_language_detection_enabled"] = processCfg.GetBool("language_detection.enabled")
 }
 
 func (ia *inventoryagent) fetchSystemProbeMetadata() {
-	// If the system-probe configuration is not loaded we fallback on zero value for all metadata
-	getBoolSysProbe := func(_ string) bool { return false }
-	getIntSysProbe := func(_ string) int { return 0 }
-
+	var sysProbeConf configGetter
 	localSysProbeConf, isSet := ia.sysprobeConf.Get()
-	if isSet {
+	if isSet && ia.conf.GetBool("system_probe_config.enabled") {
```

Comment:
You can have the system probe running even with this config disabled, especially if you have `runtime_security_config.enabled`